apps/api/github.py

```diff
@@ -0,0 +1,148 @@
+        with statsd.timer('github.{}'.format(msg)):
+            data['context'] = 'addons/linter'
+            log.info('Body: {}'.format(data))
+            res = requests.post(
+                url,
+                json=data,
+                auth=(settings.GITHUB_API_USER, settings.GITHUB_API_TOKEN))
+            log.info('Response: {}'.format(res.content))
+            res.raise_for_status()
+
+    def pending(self):
+        self.post(self.data['status_url'], data={'state': 'pending'})
+
+    def success(self):
+        self.post(self.data['status_url'], data={'state': 'success'})
+
+    def error(self):
+        self.post(self.data['status_url'], data={
+            'state': 'error',
+            # Not localising because we aren't sure what locale to localise to.
+            'description': 'This add-on did not validate.'
+        })
+
+    def failure(self):
+        self.post(self.data['status_url'], data={
+            'state': 'failure',
+            # Not localising because we aren't sure what locale to localise to.
+            'description': 'The validator failed to run correctly.'
+        })
+
+    def comment(self, data):
+        self.post(self.data['comment_url'], data=data)
+
+
+class GithubRequest(forms.Form):
+    comment_url = forms.URLField(required=False)
+    status_url = forms.URLField(required=False)
+    zip_url = forms.URLField(required=False)
+    sha = forms.CharField(required=False)
+
+    def get_status(self):
+        return (
+            self.data['pull_request']['head']['repo']['statuses_url']
+            .replace('{sha}', self.data['pull_request']['head']['sha'])
+        )
+
+    def get_zip(self):
+        return (
+            self.data['pull_request']['head']['repo']['archive_url']
+            .replace('{archive_format}', 'zipball')
+            .replace('{/ref}', '/' + self.data['pull_request']['head']['sha'])
+        )
+
+    def get_comment(self):
+        number = str(self.data['pull_request']['number'])
+        return (
+            self.data['pull_request']['head']['repo']['pulls_url']
+            .replace('{/number}', '/' + number) + '/comments')
+
+    def validate_url(self, url):
+        if not url.startswith('https://api.github.com/'):
+            raise forms.ValidationError('Invalid URL: {}'.format(url))
+        return url
+
+    def clean(self):
+        fields = (
+            ('status_url', self.get_status),
+            ('zip_url', self.get_zip),
+            ('comment_url', self.get_comment)
+        )
+        for url, method in fields:
+            try:
+                self.cleaned_data[url] = self.validate_url(method())
+            except:
+                log.error('Invalid data in processing JSON')
+                raise forms.ValidationError('Invalid data')
+
+        self.cleaned_data['sha'] = self.data['pull_request']['head']['sha']
+        self.cleaned_data['type'] = 'github'
+        return self.cleaned_data
+
+
+def rezip_file(response, pk):
```

Comment:
Could you use `jpm` to do this instead? Unzip the file from github and run `jpm xpi` inside of it? I think it would have the same end result for non-jpm add-ons (zipping everything) and would add support for jpm add-ons which might need more than just zipping it up to get a valid add-on.